services/sharepoint-connector/src/permissions-sync/permissions-sync.service.ts

```diff
@@ -0,0 +1,97 @@
+import { FetchGroupsWithMembershipsQuery } from './fetch-groups-with-memberships.query';
+import { SyncSharepointGroupsToUniqueCommand } from './sync-sharepoint-groups-to-unique.command';
+import { UniqueGroupsMap, UniqueUsersMap } from './types';
+import { groupDistinctId } from './utils';
+
+@Injectable()
+export class PermissionsSyncService {
+  private readonly logger = new Logger(this.constructor.name);
+
+  public constructor(
+    private readonly fetchGraphPermissionsMapQuery: FetchGraphPermissionsMapQuery,
+    private readonly fetchGroupsWithMembershipsQuery: FetchGroupsWithMembershipsQuery,
+    private readonly syncSharepointGroupsToUniqueCommand: SyncSharepointGroupsToUniqueCommand,
+    private readonly uniqueGroupsService: UniqueGroupsService,
+    private readonly uniqueUsersService: UniqueUsersService,
+  ) {}
+
+  public async syncPermissionsForSite(
+    siteId: string,
+    items: SharepointContentItem[],
+  ): Promise<void> {
+    const logPrefix = `[SiteId: ${siteId}]`;
+    this.logger.log(`${logPrefix} Starting permissions fetching for ${items.length} items`);
+    const permissionsFetchStartTime = Date.now();
+    const permissionsMap = await this.fetchGraphPermissionsMapQuery.run(siteId, items);
+    this.logger.log(
+      `${logPrefix} Fetched permissions for ${items.length} items in ${elapsedSecondsLog(permissionsFetchStartTime)}`,
+    );
+
+    const uniqueGroupPermissions = pipe(
+      permissionsMap,
+      values(),
+      flat(),
+      filter((permission) => permission.type !== 'user'),
+      uniqueBy(groupDistinctId),
+    );
+    this.logger.log(
+      `${logPrefix} Fetching groups with memberships from SharePoint & Graph APIs for ` +
+        `${uniqueGroupPermissions.length} unique group permissions`,
+    );
+    const groupsWithMemberships = await this.fetchGroupsWithMembershipsQuery.run(
+      siteId,
+      uniqueGroupPermissions,
+    );
+    this.logger.log(`${logPrefix} Groups with memberships fetched successfully`);
+
+    this.logger.log(
+      `${logPrefix} Found ${Object.keys(groupsWithMemberships).length} groups with memberships`,
+    );
+
+    const uniqueUsersMap = await this.getUniqueUsersMap();
+    const uniqueGroupsMap = await this.getUniqueGroupsMap();
+
+    this.logger.log(
+      `${logPrefix} Found ${Object.keys(uniqueGroupsMap).length} unique groups and ${Object.keys(uniqueUsersMap).length} unique users`,
+    );
+
+    const { updatedUniqueGroupsMap } = await this.syncSharepointGroupsToUniqueCommand.run({
+      sharePointGroupsMap: groupsWithMemberships,
+      uniqueGroupsMap,
+      uniqueUsersMap,
+    });
+
+    this.logger.log(
+      `${logPrefix} Synced ${Object.keys(updatedUniqueGroupsMap).length} resulting unique groups`,
+    );
+  }
+
+  private async getUniqueUsersMap(): Promise<UniqueUsersMap> {
+    return pipe(
+      await this.uniqueUsersService.listAllUsers(),
+      indexBy(prop('email')),
+      mapValues(prop('id')),
+    );
+  }
```

Comment:
Don't change anything, This is only reviewer help to easier understand what's happening until I get familiar with remada 
```suggestion
private async getUniqueUsersMap(): Promise<UniqueUsersMap> {
  const users = await this.uniqueUsersService.listAllUsers();
  return Object.fromEntries(users.map(user => [user.email, user.id]));
}
```